AEPMedia/Sources/MediaCollection/MediaRealTimeSession.swift

```diff
@@ -178,18 +179,59 @@ class MediaRealTimeSession: MediaSession {
     }
 
     /// Handles if there is an error in sending hit to the Media Collection Server
-    private func handleProcessingError(sessionStart: Bool) {
-        if !sessionStart || sessionStartRetryCount >= MediaRealTimeSession.MAX_ALLOWED_FAILURE {
+    private func handleProcessingError(eventType: String, connection: HttpConnection? = nil) {
+        if !shouldRetry(eventType: eventType, connection: connection) {
             sendNextHit()
             return
         }
 
-        if sessionStartRetryCount < MediaRealTimeSession.MAX_ALLOWED_FAILURE {
-            sessionStartRetryCount += 1
-            dispatchQueue.asyncAfter(deadline: .now() + .seconds(retryDuration)) { [weak self] in
-                self?.trySendHit()
+        sessionStartRetryCount += 1
+        dispatchQueue.asyncAfter(deadline: .now() + .seconds(retryDuration)) { [weak self] in
+            self?.trySendHit()
+        }
+    }
+
+    private func shouldRetry(eventType: String, connection: HttpConnection?) -> Bool {
+        guard eventType == MediaConstants.MediaCollection.EventType.SESSION_START else {
+            Log.trace(label: Self.LOG_TAG, "[\(Self.CLASS_NAME)<\(#function)>] - [Session (\(self.id))] \(eventType) request resulted in error and since event type is not sessionStart it will be dropped.")
+            return false
+        }
+
+        guard sessionStartRetryCount < MediaRealTimeSession.MAX_ALLOWED_FAILURE else {
+            Log.trace(label: Self.LOG_TAG, "[\(Self.CLASS_NAME)<\(#function)>] - [Session (\(self.id))] \(eventType) request resulted in error but since it surpassed the retry limit:(\(MediaRealTimeSession.MAX_ALLOWED_FAILURE), the request will be dropped.")
+            return false
+        }
+
+        guard let connection = connection else {
+            // should reach here when no connection (http, url) error
+            // retry for case when error occured while parsing sessionStart response for sessionId
+            Log.trace(label: Self.LOG_TAG, "[\(Self.CLASS_NAME)<\(#function)>] - [Session (\(self.id))] Unable to extract sessionId from sessionStart request response, the request will be retried.")
```

Comment:
nit: maybe state the connection is nil here in the logs, such as "... from sessionStart request response as the connection is nil, the request will be retried."

Reply:
Connection is never nil. We invoke error from sendHit when we failed to get or the parse the sessionId from the server even when network request was successful with 200. So this error also should not occur in ideal situation.